robottelo/helpers.py

```diff
@@ -750,3 +751,218 @@ def host_provisioning_check(ip_addr):
         raise ProvisioningCheckError(
             'Failed to ping virtual machine Error:{0}'.format(
                 result.stdout))
+
+
+# --- Issue based Pytest markers ---
+
+def is_open(issue, data=None):
+    """Check if specific issue is open.
+
+    Issue must be prefixed by its handler e.g:
+
+    Bugzilla: BZ:123456
+
+    Arguments:
+        issue {str} -- A string containing handler + number e.g: BZ:123465
+        data {dict} -- Issue data indexed by <handler>:<number> or None
+    """
+    # Handlers can be extended to support different issue trackers.
+    handlers = {'BZ': bugzilla.is_open_bz}
+    supported_handlers = tuple(f"{handler}:" for handler in handlers.keys())
+
+    if str(issue).startswith(supported_handlers):
+        handler_code, _, _ = str(issue).partition(":")
+    else:  # EAFP
+        raise AttributeError(
+            "is_open argument must be a string starting with a handler code "
+            "e.g: 'BZ:123456'"
+            f"supported handlers are: {supported_handlers}"
+
+        )
+    return handlers[handler_code.strip()](issue.strip(), data)
+
+
+def _should_deselect(issue, data=None):
+    """Check if test should be deselected based on marked issue."""
+    # Handlers can be extended to support different issue trackers.
+    handlers = {'BZ': bugzilla.should_deselect_bz}
+    supported_handlers = tuple(f"{handler}:" for handler in handlers.keys())
+    if str(issue).startswith(supported_handlers):
+        handler_code, _, _ = str(issue).partition(":")
+        return handlers[handler_code.strip()](issue.strip(), data)
```

Comment:
I thought about that, but at the end chose to go with 2 separate functions, because one is public, exposed to be imported and other is to be used only by the the `generate_issue_collection`, also the first is throwing error if can't handle the issue, the other is just returning None.

Well maybe it falls on [YAGNI](https://www.martinfowler.com/bliki/Yagni.html) but for example, if we decide to use github issues to mark tests, maybe github is not deselecting tests.

Reply:
fair enough.